### PR TITLE
Explore: Add ability to include tags in trace to metrics queries

### PR DESCRIPTION
**Pull request by connorlindsey**
**What this PR does / why we need it**:
Adds tags to trace to metrics config
Interpolates tags in queries

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/datasources/tempo.md [hunk 1]
```diff
@@ -46,11 +46,12 @@ This is a configuration for the [trace to logs feature]({{< relref "../explore/t
 To configure trace to metrics, select the target Prometheus data source and create any desired linked queries.
 
 -- **Data source -** Target data source.
+-- **Tags -** The tags that may be used in the linked queries. The key is the span attribute name. The optional value is the corresponding metric label name (e.g. map `k8s.pod` to `pod`). You may interpolate these tags into your queries using the `$__tags` keyword.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:49]:
```suggestion
-- **Tags -** You can use tags in the linked queries. The key is the span attribute name. The optional value is the corresponding metric label name (for example, map `k8s.pod` to `pod`). You can interpolate tags into your queries using the `$__tags` keyword.
```

File: docs/sources/datasources/tempo.md [hunk 2]
```diff
@@ -46,11 +46,12 @@ This is a configuration for the [trace to logs feature]({{< relref "../explore/t
...
 
 Each linked query consists of:
 
 -- **Link Label -** (Optional) Descriptive label for the linked query.
--- **Query -** Query that runs when navigating from a trace to the metrics data source.
+-- **Query -** Query that runs when navigating from a trace to the metrics data source. Interpolate tags using the `$__tags` keyword. For example, with the tags `k8s.pod=pod` and `cluster` configured the query `requests_total{$__tags}` would result in `requests_total{pod="nginx-554b9", cluster="us-east-1"}`.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/tempo.md:54]:
```suggestion
-- **Query -** Query that runs when navigating from a trace to the metrics data source. Interpolate tags using the `$__tags` keyword. For example, when you configue the query `requests_total{$__tags}`with the tags `k8s.pod=pod` and `cluster`, it results in `requests_total{pod="nginx-554b9", cluster="us-east-1"}`.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Comment by connorlindsey** [comment]
Thanks @achatterjee-grafana, applied those suggestions!

**Review by achatterjee-grafana — APPROVED** [review]

File: public/app/features/explore/TraceView/createSpanLink.tsx [hunk 3]
```diff
@@ -362,3 +361,27 @@ function getTimeRangeFromSpan(
...
 }
+
+// Interpolates span attributes into trace to metric query, or returns default query
+function buildMetricsQuery(query: TraceToMetricQuery, tags: Array<KeyValue<string>> = [], span: TraceSpan): string {
+  if (!query.query) {
+    return `histogram_quantile(0.5, sum(rate(tempo_spanmetrics_latency_bucket{operation="${span.operationName}"}[5m])) by (le))`;
+  }
+
+  let expr = query.query;
+  if (tags.length && query.query.indexOf('$__tags') !== -1) {
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/createSpanLink.tsx:372]: can change query.query to expr

File: public/app/features/explore/TraceView/createSpanLink.tsx [hunk 4]
```diff
@@ -362,3 +361,27 @@ function getTimeRangeFromSpan(
     },
   };
 }
+
+// Interpolates span attributes into trace to metric query, or returns default query
+function buildMetricsQuery(query: TraceToMetricQuery, tags: Array<KeyValue<string>> = [], span: TraceSpan): string {
+  if (!query.query) {
+    return `histogram_quantile(0.5, sum(rate(tempo_spanmetrics_latency_bucket{operation="${span.operationName}"}[5m])) by (le))`;
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/createSpanLink.tsx:368]: I'm curious why this is the query that's used if we don't have a query..
1. why this query
2. under what circumstances would we not have a query (the query.query type is not optional)

**Comment by joey-grafana** [comment]
An issue with wrapping a tag with a dot in quotes? (1st + 2nd image)

Screenshot 2022-05-25 at 13 17 16
Screenshot 2022-05-25 at 13 17 07


An issue with tag: value pairing? (3rd + 4th image)

Screenshot 2022-05-25 at 13 17 56
Screenshot 2022-05-25 at 13 18 04

File: public/app/features/explore/TraceView/createSpanLink.tsx (hunk 4 above)
Comment (connorlindsey) [reply public/app/features/explore/TraceView/createSpanLink.tsx:368]: 1. It's a reasonable default if users are generating span metrics and shows the average latency for that particular span. 
2. It'd be pretty uncommon, but it'll get used if the query is blank

**Comment by connorlindsey** [comment]
@joey-grafana 1/2. Prometheus can't have labels with periods so that's correctly erroring. That's why we have the option to map span attribute names (service.name) to Prometheus labels (service). 

3/4. That's working correctly. See above - you mapped the service.name attribute to be the `lb` label. Note, when you set the tags in the config, you're selecting the tag names, not the values. The values are dynamically inserted based on the span at runtime.

File: public/app/features/explore/TraceView/createSpanLink.tsx (hunk 4 above)
Comment (connorlindsey) [reply public/app/features/explore/TraceView/createSpanLink.tsx:368]: I'll make the query.query optional to indicate that.

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**